### PR TITLE
feat(tags): add resource tags to G-R services

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Context

Add resource tags to G-R services:
- [x] - guardduty
- [x] - glacier
- [x] - glue (N/A)
- [x] - iam
- [x] - kms
- [x] - macie (N/A)
- [x] - opensearch
- [x] - rds
- [x] - redshift
- [x] - route53

### Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/iam/iam_service.py [hunk 1]
```diff
@@ -408,6 +409,20 @@ def __list_server_certificates__(self):
         finally:
             return server_certificates
 
+    def __list_tags_for_resource__(self):
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_service.py:412]: I think is better to split this function into two, one for users and the other for roles.

File: prowler/providers/aws/services/rds/rds_service.py [hunk 2]
```diff
@@ -95,11 +96,13 @@ def __describe_db_snapshots__(self, regional_client):
                         )
                     ):
                         if snapshot["Engine"] != "docdb":
+                            print(snapshot)
```
Comment (jfagoagas) [inline prowler/providers/aws/services/rds/rds_service.py]:
```suggestion
```

File: prowler/providers/aws/services/rds/rds_service.py [hunk 3]
```diff
@@ -139,11 +142,13 @@ def __describe_db_cluster_snapshots__(self, regional_client):
...
```
Comment (jfagoagas) [inline prowler/providers/aws/services/rds/rds_service.py]:
```suggestion
```

File: tests/providers/aws/services/guardduty/guardduty_is_enabled/guardduty_is_enabled_test.py [hunk 4]
```diff
@@ -33,6 +33,7 @@ def test_guardduty_enabled(self):
             Detector(
                 id=detector_id,
                 region=AWS_REGION,
+                arn="",
```
Comment (jfagoagas) [inline tests/providers/aws/services/guardduty/guardduty_is_enabled/guardduty_is_enabled_test.py]: Can you fill this value?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/iam/iam_service.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/iam/iam_service.py:412]: Done

File: prowler/providers/aws/services/rds/rds_service.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/rds/rds_service.py]: Done

File: prowler/providers/aws/services/rds/rds_service.py (hunk 3 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/rds/rds_service.py]: Done

File: tests/providers/aws/services/guardduty/guardduty_is_enabled/guardduty_is_enabled_test.py (hunk 4 above)
Comment (MrCloudSec) [reply tests/providers/aws/services/guardduty/guardduty_is_enabled/guardduty_is_enabled_test.py]: Done

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**